src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer machineProducer;
+        if (Bulldozer.class.equals(type)) {
+            machineProducer = new BulldozerProducer();
+            return machineProducer.get();
+        }
+        if (Truck.class.equals(type)) {
+            machineProducer = new TruckProducer();
+            return machineProducer.get();
+        }
+        if (Excavator.class.equals(type)) {
+            machineProducer = new ExcavatorProducer();
+            return machineProducer.get();
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, T value) {
+        int size = machines.size();
+        for (int i = 0; i < size; i++) {
+            machines.set(i, value);
```

Comment:
Here you can use Collections.fill(machines, value);
```suggestion
        Collections.fill(machines, value);
```